crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -170,59 +1066,204 @@ pub enum RefundStatus {
+            JpmorganTransactionState::Declined | JpmorganTransactionState::Error => Self::Failure,
+            JpmorganTransactionState::Pending | JpmorganTransactionState::Authorized => {
+                Self::Pending
+            }
+        }
+    }
+}
+
+impl TryFrom<RefundsResponseRouterData<Execute, JpmorganRefundResponse>>
+    for RefundsRouterData<Execute>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: RefundsResponseRouterData<Execute, RefundResponse>,
+        item: RefundsResponseRouterData<Execute, JpmorganRefundResponse>,
     ) -> Result<Self, Self::Error> {
+        let refund_id = item
+            .response
+            .transaction_id
+            .clone()
+            .ok_or(errors::ConnectorError::ResponseHandlingFailed)?;
+
+        let transaction_state = item.response.transaction_state;
+        let refund_status = common_enums::RefundStatus::from_transaction_state(transaction_state);
         Ok(Self {
             response: Ok(RefundsResponseData {
-                connector_refund_id: item.response.id.to_string(),
-                refund_status: enums::RefundStatus::from(item.response.status),
+                connector_refund_id: refund_id,
+                refund_status,
             }),
             ..item.data
         })
     }
 }
 
-impl TryFrom<RefundsResponseRouterData<RSync, RefundResponse>> for RefundsRouterData<RSync> {
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganRefundSyncResponse {
+    transaction_id: String,
+    request_id: String,
+    transaction_state: JpmorganTransactionState,
+    amount: MinorUnit,
+    currency: common_enums::Currency,
+    response_status: JpmorganResponseStatus,
+    response_code: String,
+}
+
+impl TryFrom<RefundsResponseRouterData<RSync, JpmorganRefundSyncResponse>>
+    for RefundsRouterData<RSync>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: RefundsResponseRouterData<RSync, RefundResponse>,
+        item: RefundsResponseRouterData<RSync, JpmorganRefundSyncResponse>,
     ) -> Result<Self, Self::Error> {
+        let refund_id = item.response.transaction_id.clone();
+        let transaction_state = item.response.transaction_state;
+        let status = common_enums::RefundStatus::from_transaction_state(transaction_state);
         Ok(Self {
             response: Ok(RefundsResponseData {
-                connector_refund_id: item.response.id.to_string(),
-                refund_status: enums::RefundStatus::from(item.response.status),
+                connector_refund_id: refund_id,
+                refund_status: status,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganCancelRequest {
+    pub amount: Option<i64>,
+    pub is_void: Option<bool>,
```

Comment:
Why is this field made optional?

Reply:
that field is marked as optional in JP Morgan's docs